benchmarks/logreg_l2/solvers/julia_gd.jl

```diff
@@ -0,0 +1,68 @@
+using Core
+using LinearAlgebra
+
+# TODO : import shared functions from file
+# include("./benchmarks/logreg_l2/solvers/logistic.jl")
+
+##########################################################""
+
+# Loss evaluation
+function logistic_loss(X, y, w::Array{Float64}) # correct
+    # return sum( log.(1. .+  exp.(-(y .* (X*w)))) ) # other implementation
+    return -sum( log.(logistic_diff_phi(y .* (X*w))) )
+end
+
+# Gradient evaluation
+function logistic_diff_phi(z::Array{Float64}) # correct
+    # Derivative of phi_i is ``more or less" the sigmoid
```

Comment:
just wondering, can you clarify what "more or less" means ? it seems to me that you compute exactly that, in 2 different ways depending on the sign of the argument, to avoid precision errors I believe ?

Reply:
I'm sorry, the function is *exactly* the sigmoid function. The "more or less" was referring to the name of the function itsfelf. Let 
<img src="https://render.githubusercontent.com/render/math?math=\phi_i (z) = \log \left( 1 %2B e^{-y_i z} \right)"> 
be the i-th loss, then its derivative is 
<img src="https://render.githubusercontent.com/render/math?math=\phi_i^' (z) = -y_i \sigma(-y_i z)"> 
To make everything clearer I'll rename this function as "sigmoid".

And you are correct, the sigmoid is computed differently based on the sign of the input to avoid numerical overflow as explained in this [post](https://medium.com/@shaktiwadekar/how-to-avoid-numerical-overflow-in-sigmoid-function-numerically-stable-sigmoid-function-5298b14720f6).